routing/single_vehicle_world_graph.hpp

```diff
@@ -29,53 +31,82 @@ class SingleVehicleWorldGraph final : public WorldGraph
   WorldGraphMode GetMode() const override { return m_mode; }
+
   void GetOutgoingEdgesList(Segment const & segment, std::vector<SegmentEdge> & edges) override;
   void GetIngoingEdgesList(Segment const & segment, std::vector<SegmentEdge> & edges) override;
+
   RouteWeight HeuristicCostEstimate(Segment const & from, Segment const & to) override;
   RouteWeight HeuristicCostEstimate(m2::PointD const & from, m2::PointD const & to) override;
   RouteWeight HeuristicCostEstimate(Segment const & from, m2::PointD const & to) override;
+
   RouteWeight CalcSegmentWeight(Segment const & segment) override;
   RouteWeight CalcLeapWeight(m2::PointD const & from, m2::PointD const & to) const override;
   RouteWeight CalcOffroadWeight(m2::PointD const & from, m2::PointD const & to) const override;
   double CalcSegmentETA(Segment const & segment) override;
+
   std::vector<Segment> const & GetTransitions(NumMwmId numMwmId, bool isEnter) override;
 
+  void SetRoutingOptions(RoutingOptions routingOptions) override { m_avoidRoutingOptions = routingOptions; }
   /// \returns true if feature, associated with segment satisfies users conditions.
   bool IsRoutingOptionsGood(Segment const & segment) override;
   RoutingOptions GetRoutingOptions(Segment const & segment) override;
-  void SetRoutingOptions(RoutingOptions routingOptions) override { m_avoidRoutingOptions = routingOptions; }
 
   std::unique_ptr<TransitInfo> GetTransitInfo(Segment const & segment) override;
   std::vector<RouteSegment::SpeedCamera> GetSpeedCamInfo(Segment const & segment) override;
 
-  // This method should be used for tests only
-  IndexGraph & GetIndexGraphForTests(NumMwmId numMwmId)
+  IndexGraph & GetIndexGraph(NumMwmId numMwmId) override
   {
     return m_loader->GetIndexGraph(numMwmId);
   }
 
-  IndexGraph & GetIndexGraph(NumMwmId numMwmId) override
+  void SetAStarParents(bool forward, ParentSegments & parents) override;
+  void SetAStarParents(bool forward, ParentJoints & parents) override;
+
+  bool AreWavesConnectible(ParentSegments & forwardParents, Segment const & commonVertex,
+                           ParentSegments & backwardParents,
+                           std::function<uint32_t(Segment const &)> && fakeFeatureConverter) override;
+  bool AreWavesConnectible(ParentJoints & forwardParents, JointSegment const & commonVertex,
+                           ParentJoints & backwardParents,
+                           std::function<uint32_t(JointSegment const &)> && fakeFeatureConverter) override;
+  // @}
+
+  // This method should be used for tests only
+  IndexGraph & GetIndexGraphForTests(NumMwmId numMwmId)
   {
     return m_loader->GetIndexGraph(numMwmId);
   }
 
 private:
+  /// \brief Get parents' featureIds of |commonVertex| from forward AStar wave and join them with
+  ///        parents' featureIds from backward wave.
+  /// \return The result chain of fetureIds used to find restrictions on it and understand whether
```

Comment:
`///  \return The result chain of featureIds are used to find restrictions on it and understand whether`